core/src/types.ts

```diff
@@ -101,6 +101,31 @@ export interface IAnswerUpdated {
   updatedAt: BigNumber
 }
 
+export interface IL2DataRequested {
+  requestId: BigNumber
+  jobId: string
+  accId: BigNumber
+  callbackGasLimit: number
+  sender: string
+  numSubmission: number
+  /* eslint-disable  @typescript-eslint/no-explicit-any */
```

Comment:
Please let me know if I am wrong but I think that this `req` is called `data` in our `L2EndpointRequestResponse.sol`.

https://github.com/Bisonai/orakl/blob/7c20632e75856f16634c225e179ca58e22ad6e2e/contracts/src/v0.1/L2EndpointRequestResponse.sol#L18

It is defined as `bytes` which means we could use type `string`. It is much better than `any`.

Reply:
I believe it's not, in this case, it is the request object here:
```
    event DataRequested(
        uint256 indexed requestId,
        bytes32 jobId,
        uint64 indexed accId,
        uint32 callbackGasLimit,
        address indexed sender,
        uint8 numSubmission,
        Orakl.Request req
    );
```

Reply:
@KelvinThai Aha! Then how about defining data type based on that?

Reply:
Sure we can. but we just need forward the whole thing from L2 to L1, not extracting any data from it.  Do you think we really need to do this in this case?

Reply:
It is not about extracting but about clarity. I thought it was just bytes that were encoded as string, but you were saying it is a special data structure. I dont think we should extract it out of string, but if it is not a string, then it should be possible to automatically map it to custom data type.